### PR TITLE
Emit thread predicates for vectorized instructions.

### PR DESCRIPTION
**Pull request by csarofeen**

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/lower_predicate.cpp [hunk 1]
```diff
@@ -42,6 +42,33 @@ class ConditionalFromPredicateModifier : public kir::IrVisitor {
     if (expr != nullptr && expr->predicate() != nullptr) {
       // Replace expr predicate with bool conditional
       auto conditional = generateConditional(expr->predicate());
+      if (expr->predicate()->predicate_type() == PredicateType::Vectorize) {
+        // TODO: This logic doesn't seem to fit well here, for unswitch the
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/lower_predicate.cpp:46]: Maybe `UnrollPass::handle(Expr*)`, where we first create `Vectorize` predicates?

Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/lower_predicate.cpp:46]: Maybe `UnrollPass::handle(Expr*)`, where we first create `Vectorize` predicates?

**Review by naoyam — APPROVED** [review]
LGTM. I guess this just hasn't manifested as failures so far.

**Comment by csarofeen** [comment]
It did in: https://github.com/csarofeen/pytorch/pull/1421 which is where I pulled it from, but in this case it was for perf reasons where we're broadcasting writes to gmem, and just want the avoid redundant writes predicate to actually do something.

File: torch/csrc/jit/codegen/cuda/lower_predicate.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/lower_predicate.cpp:46]: I didn't find a great place to handle this yet, I was thinking the same but it didn't seem clean/easy. I'm going to leave the TODO for now.

**Pull request merged**